components/eam/cime_config/config_component.xml

```diff
@@ -71,7 +71,7 @@
       <!--  -->
       <value compset="GEOS_EAM"       >-offline_dyn</value>
       <value compset="GEOS_EAM"      >-nlev 56</value>
-      <value compset="_EAM.*_BGC%B"   >-co2_cycle</value>
+      <value compset="_EAM.*_BGC%*"   >-co2_cycle</value>
```

Comment:
I think these logic changes are safe for the compsets that we currently have implemented in the code (and so won't have unintended side effects), according to my notes here: https://acme-climate.atlassian.net/wiki/spaces/EBGC/pages/151290237/BGC+Specification+of+CO2+coupling+diagnostic+prognostic+radiative+BGC

However, I didn't double-check the code base to verify this.  Can you confirm that this was looked at?

Reply:
1. Added smoke tests just now with help from @singhbalwinder 
2. I tested this changes before PR

Reply:
I've looked through the code some and the notes @susburrows linked here, and it doesn't look like there will be any problems with this change.  

I personally like this change since I don't think prognostic atmospheric CO2 should require compset naming to begin with the letter 'B' after the _BGC tag.

Reply:
Thanks for cross-checking this, @beharrop , and for adding the smoke tests, @BunnyVon and @singhbalwinder .